src/state/Maps/selectors.js

```diff
@@ -1,1181 +1,501 @@
-	roll: 0,
-	heading: 0
-};
-
-const mapConstants = {defaultMapView};
-
-let getBackgroundLayerCache = new CacheFifo(10);
-let getLayersCache = new CacheFifo(10);
-
-/* ===== Basic selectors ==================== */
 const getSubstate = state => state.maps;
 
+const getActiveMapKey = state => state.maps.activeMapKey;
 const getMapsAsObject = state => state.maps.maps;
 const getMapSetsAsObject = state => state.maps.sets;
 
-const getActiveSetKey = state => state.maps.activeSetKey;
-const getActiveMapKey = state => state.maps.activeMapKey;
-
-const getMaps = createSelector(
-	[getMapsAsObject],
-	(maps) => {
-		if (maps && !_.isEmpty(maps)) {
-			return Object.values(maps);
-		} else {
-			return null;
-		}
-	}
-);
-
-const getMapSets = createSelector(
-	[getMapSetsAsObject],
-	(sets) => {
-		if (sets && !_.isEmpty(sets)) {
-			return Object.values(sets);
-		} else {
-			return null;
-		}
-	}
-);
-
 /**
  * @param state {Object}
  * @param mapKey {string}
  */
-const getMapByKey = createCachedSelector(
-	[getMapsAsObject,
-	(state, key) => key],
-	(maps, key) => {
-		if (maps && !_.isEmpty(maps) && key) {
-			return maps[key] ? maps[key] : null;
-		} else {
-			return null;
-		}
-	}
-)((state, key) => key);
-
-
-const getMapLayersByMapKey = createSelector(
-	[getMapByKey],
-	(map) => {
-		return map && map.data && map.data.layers || null;
-	}
+const getMapByKey = createSelector(
+    [
+        getMapsAsObject,
+        (state, key) => key
+    ],
+    (maps, key) => {
+        return maps?.[key] || null;
+    }
 );
 
-
+/**
+ * @param state {Object}
+ */
+const getMapSets = createSelector(
+    [getMapSetsAsObject],
+    (sets) => {
+        if (sets && !_.isEmpty(sets)) {
+            return Object.values(sets);
+        } else {
+            return null;
+        }
+    }
+);
 
 /**
  * @param state {Object}
  * @param setKey {string}
  */
 const getMapSetByKey = createSelector(
-	[getMapSetsAsObject,
-	(state, key) => key],
-	(sets, key) => {
-		if (sets && !_.isEmpty(sets) && key) {
-			return sets[key] ? sets[key] : null;
-		} else {
-			return null;
-		}
-	}
+    [
+        getMapSetsAsObject,
+        (state, key) => key
+    ],
+    (sets, key) => {
+        return sets?.[key] || null;
+    }
 );
 
 /**
  * @param state {Object}
  * @param mapKey {string}
  */
 const getMapSetByMapKey = createSelector(
-	[getMapSets,
-	(state, mapKey) => (mapKey)],
-	(sets, mapKey) => {
-		if (sets && !_.isEmpty(sets) && mapKey) {
-			let setForMap = null;
-			sets.forEach((set) => {
-				if (set.maps && _.includes(set.maps, mapKey)) {
-					setForMap = set;
-				}
-			});
-			return setForMap;
-		} else {
-			return null;
-		}
-	}
+    [
+        getMapSets,
+        (state, mapKey) => (mapKey)
+    ],
+    (sets, mapKey) => {
+        if (sets && !_.isEmpty(sets) && mapKey) {
+            let setForMap = null;
+            sets.forEach((set) => {
+                if (set.maps && _.includes(set.maps, mapKey)) {
+                    setForMap = set;
```

Comment:
The change you propose is not so readable IMHO

Reply:
Problem is, that you are iterating over all sets every time. In case of use .find iterating stops when the condition is fulfilled.

I know, usually is just a few sets. But the pattern is not good.